src/fio/fioPlugin.js

```diff
@@ -0,0 +1,217 @@
+    )
+    const valid = checkAddress(edgeParsedUri.publicAddress || '')
+    if (!valid) {
+      throw new Error('InvalidPublicAddressError')
+    }
+    if (parsedUri.query.memo_type) {
+      if (parsedUri.query.memo_type === 'MEMO_ID') {
+        if (parsedUri.query.memo) {
+          const m = bns.add(parsedUri.query.memo, '0')
+          // Check if the memo is an integer
+          if (m !== parsedUri.query.memo) {
+            throw new Error('ErrorInvalidMemoId')
+          }
+          edgeParsedUri.uniqueIdentifier = parsedUri.query.memo
+        }
+      }
+    }
+    return edgeParsedUri
+  }
+
+  async encodeUri (obj: EdgeEncodeUri): Promise<string> {
+    const valid = checkAddress(obj.publicAddress)
+    if (!valid) {
+      throw new Error('InvalidPublicAddressError')
+    }
+    let amount
+    if (typeof obj.nativeAmount === 'string') {
+      const currencyCode: string = 'FIO'
+      const nativeAmount: string = obj.nativeAmount
+      const denom = getDenomInfo(currencyInfo, currencyCode)
+      if (!denom) {
+        throw new Error('InternalErrorInvalidCurrencyCode')
+      }
+      amount = bns.div(nativeAmount, denom.multiplier, 16)
+    }
+    const encodedUri = this.encodeUriCommon(obj, 'fio', amount)
+    return encodedUri
+  }
+
+  async getAccSystemStats (account: string) {
+    return new Promise((resolve, reject) => {
+      this.eosServer.getAccount(account, (error, result) => {
+        if (error) {
+          if (error.message.includes('unknown key')) {
+            error.code = 'ErrorUnknownAccount'
+          }
+          reject(error)
+        }
+        resolve(result)
+      })
+    })
+  }
+}
+
+export function makeFioPlugin (opts: EdgeCorePluginOptions): EdgeCurrencyPlugin {
+  const { io } = opts
+  const fetchJson = getFetchJson(opts)
+
+  let toolsPromise: Promise<FioPlugin>
+  function makeCurrencyTools (): Promise<FioPlugin> {
+    if (toolsPromise != null) return toolsPromise
+    toolsPromise = Promise.resolve(new FioPlugin(io))
+    return toolsPromise
+  }
+
+  async function makeCurrencyEngine (
+    walletInfo: EdgeWalletInfo,
+    opts: EdgeCurrencyEngineOptions
+  ): Promise<EdgeCurrencyEngine> {
+    const tools = await makeCurrencyTools()
+    const currencyEngine = new FioEngine(tools, walletInfo, opts, fetchJson)
+    await currencyEngine.loadEngine(tools, walletInfo, opts)
+
+    currencyEngine.otherData = currencyEngine.walletLocalData.otherData
+    // currencyEngine.otherData is an opaque utility object for use for currency
+    // specific data that will be persisted to disk on this one device.
+    // Commonly stored data would be last queried block height or nonce values for accounts
+    // Edit the flow type EosWalletOtherData and initialize those values here if they are
+    // undefined
+    // TODO: Initialize anything specific to this currency
+    // if (!currencyEngine.otherData.nonce) currencyEngine.otherData.nonce = 0
+    if (!currencyEngine.otherData.accountName) {
```

Comment:
Since these are properties of the currency engine, it seems like they should be set up in the currency engine constructor. Assuming they are even needed (they look like more EOS leftovers)?